arangod/Agency/Agent.cpp

```diff
@@ -1200,14 +1200,15 @@ read_ret_t Agent::read(query_t const& query) {
     return read_ret_t(false, NO_LEADER);
   }
 
-  std::string leaderId = _constituent.leaderID();
+  leader = _constituent.leaderID();
```

Comment:
why do we have to relead `leader` at all?

Reply:
That re-reading of the leader was introduced in 22e46978a6ce38d69fd5519188824b7e2dd67037 by @neunhoef. Please check with him.